transaction_integration_test.go

```diff
@@ -725,3 +726,185 @@ func TestTransactionCreateWithCustomFields(t *testing.T) {
+		Amount: amount,
+		CreditCard: &CreditCard{
+			Number:         testCreditCards["visa"].Number,
+			ExpirationDate: "05/14",
+		},
+		MerchantAccountId: testSubMerchantAccountId,
+		ServiceFeeAmount:  amount,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	id := txn.Id
+	txn, err = testGateway.Transaction().HoldInEscrow(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if txn.EscrowStatus != EscrowStatus.HoldPending {
+		t.Fatalf("expected EscrowStatus to be %s, was %s", EscrowStatus.HoldPending, txn.EscrowStatus)
+	}
+}
+
+func TestReleaseFromEscrow(t *testing.T) {
+	testSubMerchantAccountId := getSubMerchantAccount(t)
+	amount := NewDecimal(6400, 2)
+	txn, err := testGateway.Transaction().Create(&Transaction{
+		Type:   "sale",
+		Amount: amount,
+		CreditCard: &CreditCard{
+			Number:         testCreditCards["visa"].Number,
+			ExpirationDate: "05/14",
+		},
+		MerchantAccountId: testSubMerchantAccountId,
+		ServiceFeeAmount:  amount,
+		Options: &TransactionOptions{
+			SubmitForSettlement: true,
+			HoldInEscrow:        true,
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	id := txn.Id
+	// _, err = escrow(id)
+	err = settle(t, id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	txn, err = testGateway.Transaction().ReleaseFromEscrow(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if txn.EscrowStatus != EscrowStatus.ReleasePending {
+		t.Fatalf("expected EscrowStatus to be %s, was %s", EscrowStatus.ReleasePending, txn.EscrowStatus)
+	}
+}
+
+func TestCancelRelease(t *testing.T) {
+	testSubMerchantAccountId := getSubMerchantAccount(t)
+	amount := NewDecimal(6500, 2)
+	txn, err := testGateway.Transaction().Create(&Transaction{
+		Type:   "sale",
+		Amount: amount,
+		CreditCard: &CreditCard{
+			Number:         testCreditCards["visa"].Number,
+			ExpirationDate: "05/14",
+		},
+		MerchantAccountId: testSubMerchantAccountId,
+		ServiceFeeAmount:  amount,
+		Options: &TransactionOptions{
+			SubmitForSettlement: true,
+			HoldInEscrow:        true,
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	id := txn.Id
+	err = settle(t, id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	txn, err = testGateway.Transaction().ReleaseFromEscrow(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if txn.EscrowStatus != EscrowStatus.ReleasePending {
+		t.Fatalf("expected EscrowStatus to be %s, was %s", EscrowStatus.ReleasePending, txn.EscrowStatus)
+	}
+	txn, err = testGateway.Transaction().CancelRelease(id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if txn.EscrowStatus != EscrowStatus.Held {
+		t.Fatalf("expected EscrowStatus to be %s, was %s", EscrowStatus.Held, txn.EscrowStatus)
+	}
+}
+
+func settle(t *testing.T, id string) error {
+	txn, err := testGateway.Transaction().Settle(id)
+	if err != nil {
+		return err
+	}
+	if txn.Status != "submitted_for_settlement" && txn.Status != "settling" && txn.Status != "settled" {
+		t.Fatalf("expected Status to be submitted_for_settlement, settling, or settled, was %s", txn.Status)
+	}
+	return nil
+}
+
+var subMerchantAccountID string
```

Comment:
I like the idea of caching this so we don't create more sub-merchants unnecessarily. If these were running serially it might make a bit difference, but I'm not sure it will for us because we run the tests in parallel. If you run the tests like `go test -parallel 15` do you see much difference with the caching used or not?